### ISSUE TITLE
Typing custom text that ends with a composed character freezes the test

### ISSUE THREAD

**Issue by rfblock** [monkeytypegame/monkeytype#3839]
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

Yes

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes

### Account name

Stegg2020

### Account config

{"theme":"chaos_theory","themeLight":"serika","themeDark":"serika_dark","autoSwitchTheme":false,"customTheme":false,"customThemeColors":["#323437","#e2b714","#e2b714","#646669","#000000","#d1d0c5","#ca4754","#7e2a33","#ca4754","#7e2a33"],"favThemes":[],"showKeyTips":true,"showLiveWpm":true,"showTimerProgress":true,"smoothCaret":true,"quickRestart":"tab","punctuation":false,"numbers":false,"words":200,"time":15,"mode":"custom","quoteLength":[0,1,2,3],"language":"english","fontSize":1.5,"freedomMode":false,"difficulty":"normal","blindMode":false,"quickEnd":false,"caretStyle":"default","paceCaretStyle":"default","flipTestColors":false,"layout":"default","funbox":"none","confidenceMode":"off","indicateTypos":"off","timerStyle":"text","colorfulMode":true,"randomTheme":"off","timerColor":"black","timerOpacity":"0.25","stopOnError":"off","showAllLines":true,"keymapMode":"react","keymapStyle":"staggered","keymapLegendStyle":"blank","keymapLayout":"qwerty","keymapShowTopRow":"never","fontFamily":"Roboto_Mono","smoothLineScroll":false,"alwaysShowDecimalPlaces":false,"alwaysShowWordsHistory":false,"singleListCommandLine":"manual","capsLockWarning":true,"playSoundOnError":false,"playSoundOnClick":"off","soundVolume":"0.5","startGraphsAtZero":true,"showOutOfFocusWarning":true,"paceCaret":"off","paceCaretCustomSpeed":200,"repeatedPace":true,"pageWidth":"100","chartAccuracy":true,"chartStyle":"line","minWpm":"off","minWpmCustomSpeed":100,"highlightMode":"letter","alwaysShowCPM":false,"ads":"result","hideExtraLetters":false,"strictSpace":false,"minAcc":"off","minAccCustom":90,"showLiveAcc":false,"showLiveBurst":false,"monkey":false,"repeatQuotes":"off","oppositeShiftMode":"off","customBackground":"","customBackgroundSize":"cover","customBackgroundFilter":[0,1,1,1,1],"customLayoutfluid":"qwerty#dvorak#colemak","monkeyPowerLevel":"off","minBurst":"off","minBurstCustomSpeed":100,"burstHeatmap":false,"britishEnglish":false,"lazyMode":false,"showAverage":"off","tapeMode":"off"}

### Current Behavior

Any custom text that ends with a composed character (i.e. æ/⎄ae, or ☭/⎄CCCP) will freeze the text. The rest of the website responds normally, but no input goes through the test and must be restarted

### Expected Behavior

The test should end once the final character is finished composing.

### Steps To Reproduce

1. Create a custom text that ends with a composed charater, i.e. "The quick brown ①"
2. Go through the test as you normally would
3. Verify that the last character is the composed character, and nothing else is appended
4. Press Space
5. The test should now be frozen, no characters can be typed or deleted.

### Environment

- OS: Pop! OS 20.04 LTS x86-64
- Browser: Google Chrome
- Browser Version: 104.0.5112.101 (Official Build) (64-bit)

### Anything else?

The browser console throws errors when trying to complete the test

**Comment by Miodec** [comment]
I tried putting `The quick brown ①` as the custom text and I didn't see any issues with finishing the test

**Comment by BBBNBBB** [comment]
I have a similar problem, but distinct in that it only freezes when inserting unicode characters via X11 (using Ctrl+Shift+U), but it works fine when using the compose key. 

Below is the error produced when pressing space to complete the test:
```
Uncaught TypeError: L.words.getCurrent() is undefined
    Y https://monkeytype.com/js/monkeytype.7505e352.js:1
    Z https://monkeytype.com/js/monkeytype.7505e352.js:1
    8553 https://monkeytype.com/js/monkeytype.7505e352.js:1
    jQuery 8
    8553 https://monkeytype.com/js/monkeytype.7505e352.js:1
    n https://monkeytype.com/js/monkeytype.7505e352.js:1
    9294 https://monkeytype.com/js/monkeytype.7505e352.js:1
    n https://monkeytype.com/js/monkeytype.7505e352.js:1
    i https://monkeytype.com/js/monkeytype.7505e352.js:1
    O https://monkeytype.com/js/monkeytype.7505e352.js:1
    <anonymous> https://monkeytype.com/js/monkeytype.7505e352.js:1
    <anonymous> https://monkeytype.com/js/monkeytype.7505e352.js:1
monkeytype.7505e352.js:1:233918
```

If the quick end option is used, there is no freezing, but it is not possible to end the test.

**Comment by Miodec** [comment]
Hey guys can you walk me through this issue again? I can't get any of the errors you mentioned to show up.

**Comment by BBBNBBB** [comment]
Hi, still unable to finish tests by Unicode input.

I am wondering whether I should open a separate issue for what I am experiencing, as the compose key works fine for me, just Unicode input that's not—both of these special character input methods are presented in the same way on my system, but they could interact with the website differently, hence worthy of a separate issue.

Steps to reproduce my error: (Using Version 108.0 (64-bit) of Firefox)

1. Create a custom test with `The quick »` (for example) as the text
2. Type until the last character
3. Press `Ctrl+Shift+U` then `bb` then press the Space key, which inputs the correct character
4. Press the Space key, now that the character is input
5. The errors should be in the Console

Whether or not quick end is enabled, the test should end as soon as the last character is input, as the character is correct, but it does not end regardless of that setting.
The test no longer freezes like last time, the test now only fails to end. I can delete the final character, then input it some other way (such as composing the character) and the test will complete without trouble.

Also, the error varies.

- First off, the `L.words.getCurrent()` has changed to `W.words.getCurrent()` since my first report.

- Secondly, the main block of the error has more lines if I am outside a Private window, even if cache is cleared and I log out.

- Thirdly, if I press the space key for a second time, after inputting the final character, it produces an error slightly different to the first press, and any space press after produces no error (but the test still does not end). Example from a Private window session below:

```
Uncaught TypeError: W.words.getCurrent() is undefined
    Z https://monkeytype.com/js/monkeytype.2953bb46.js:1
    X https://monkeytype.com/js/monkeytype.2953bb46.js:1
    8553 https://monkeytype.com/js/monkeytype.2953bb46.js:1
    jQuery 2
monkeytype.2953bb46.js:1:237764
```
and on second space press:
```
Uncaught TypeError: W.words.getCurrent() is undefined
    X https://monkeytype.com/js/monkeytype.2953bb46.js:1
    8553 https://monkeytype.com/js/monkeytype.2953bb46.js:1
    jQuery 2
monkeytype.2953bb46.js:1:239259
```

Note: When inputting through Alt codes under Windows, no such error is encountered.

Hope this helps.

**Comment by Miodec** [comment]
Well, this will be hard to find, because it's not showing up on Mac. Looks to be Windows specific.

**Comment by BBBNBBB** [comment]
Linux, Windows is not effected, but I see the issue.
Does anyone part of the Monkeytype team use Linux? 
I suspect it is related to IBus, which seems quite universal when it comes to Linux, but not used by Mac OS or Windows.

**Comment by Miodec** [comment]
Sadly, the "Monkeytype team" is just me lol. Contributors come and go. I might have someone who I can ask.

**Comment by rizwanmustafa** [comment]
@BBBNBBB Can you provide details about the operating system you are running it on. Like which distribution of linux, what version of the OS and the browser etc. Thanks

**Comment by BBBNBBB** [comment]
Kubuntu 22.10
KDE Plasma 5.25.5
Kernal 5.19.0-35-generic (64-bit)
Firefox 108.0 (64-bit)

**Comment by rizwanmustafa** [comment]
@BBBNBBB I am on linux and it works fine for me. Could you perhaps post a video? Thanks

**Comment by BBBNBBB** [comment]
First I input the character by Unicode input—no success, but the character does show as correct (not red).
Then I input by composing the character (⎄>>), which completes the test.

https://user-images.githubusercontent.com/110104157/224126299-ab85969c-e0de-449a-8aa3-5ec0aaeedf70.mp4

**Comment by meyerhenning** [comment]
I am on Fedora 37 Workstation and get this error too.

Custom text:
`The quick brown jumps over the lazy dog ①`

Input method:
X11, like @BBBNBBB mentioned

After entering the unicode char and pressing space:
![image](https://user-images.githubusercontent.com/82178058/227788864-046429b3-4340-4e2b-a3fe-3636219c2a5e.png)

**Comment by Miodec** [comment]
Hey guys, I think I resolved this issue while fixing #4407. Tested on a Linux VM and it looks to be working. Can you test?

**Comment by arcoshine** [comment]
It works for me now. Thanks
